boards/arc/qemu/board.cmake

```diff
@@ -0,0 +1,28 @@
+set(EMU_PLATFORM qemu)
+
+set(QEMU_CPU_TYPE_${ARCH} arc)
+
+if(${CONFIG_SOC_QEMU_EM})
+set(QEMU_CPU_TYPE_${ARCH} arcem)
+set(QEMU_FLAGS_${ARCH} -cpu arcem)
+elseif(${CONFIG_SOC_QEMU_HS})
+set(QEMU_CPU_TYPE_${ARCH} archs)
+set(QEMU_FLAGS_${ARCH} -cpu archs)
+endif()
+
+list(APPEND QEMU_FLAGS_${ARCH}
+  -M simhs
+  -m 128M
```

Comment:
Invoking Qemu with `128M` but only using SRAM.. :-(

Reply:
Well that `-m 128M` is a left-over of what we used to use for all the rest including Linux use-case etc. Here I don't know which option to chose:
1. Decrease memory size in QEMU to match 256 KiB in the Zephyr or
2. Increase amount of memory claimed in Zephyr to 128 MiB
3. Something in between

What makes more sense in case of Zephyr?

Reply:
Zephyr generally runs on low-end targets which have little Ram / Flash, and so you have no other option but to use them. However, in my opinion, this idea should not seep into platforms which have large amount of memory. In short, the bsp / board should be free to utilize the complete memory available on the board. 

There are cases where you don't want to use the complete memory - Mostly when you have multiple operating systems running on different cores of the same SoC. In these cases, you would have to "partition" the memory between both OSes.. But that's not applicable here.. 

So my opinion in this regard is to use the complete available memory.. May be others disagree ?

Reply:
Agree, the dts/linker files restrict already to a 'reasonable' size. Unless there's impact on memory/performance when configuring QEMU with larger memory, for flexbility it's fine to define the 128M here.

Reply:
128M is too much....can we drop this down to some reasonable value so that workstation ram usage doesn't blow up during highly parallel sanitycheck runs? On x86 we use 8M IIRC, there's no need for this to be so large

Reply:
@andrewboie right that makes perfect sense, I didn't consider multiple parallel sanitycheck runs on non-top notch full-blown servers :)

Reply:
@andrewboie done - down to 8 MiB in both .dts & QEMU command-line.